d2/src/main/java/com/linkedin/d2/balancer/clusterfailout/ClusterFailoutRedirectConfig.java

```diff
@@ -0,0 +1,76 @@
+package com.linkedin.d2.balancer.clusterfailout;
+
+import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class is a simple data structure for controlling the weighting of how much traffic is sent to a given fabric.
+ */
+
+public class ClusterFailoutRedirectConfig
+{
+  // Fabric that this configuration is addressing.
+  private final String _fabricUrn;
+  // Relative weight for traffic distribution
+  private final Integer _weight;
+
+  private final static String FABRIC_URN_PROPERTY = "fabric";
+  private final static String WEIGHT_PROPERTY = "weight";
+
+  private static final Logger _logger = LoggerFactory.getLogger(
+      ClusterFailoutRedirectConfig.class);
+
+  public ClusterFailoutRedirectConfig(String fabricUrn, Integer weight) {
+    _fabricUrn = fabricUrn;
+    _weight = weight;
+  }
+
+  public static ClusterFailoutRedirectConfig createFromMap(Map<String, Object> configMap) {
+    try {
+      return new ClusterFailoutRedirectConfig((String) configMap.get(ClusterFailoutRedirectConfig.FABRIC_URN_PROPERTY),
+          (Integer) configMap.get(ClusterFailoutRedirectConfig.WEIGHT_PROPERTY));
+    } catch (Exception e) {
+      _logger.error("Error while converting SLF properties: " + e.getMessage());
+      return null;
+    }
```

Comment:
we need to add tests for the normal cases and edge cases when the map doesn't have the keys, the values are null, or value type conversion errors that trigger any exceptions. 

Please see my other comment about adding the pdl and converter. If those are added, the tests can be added in the converter test (similar to [anaryDistributionStrategyConverterTest.java](https://github.com/linkedin/rest.li/pull/755/files#diff-5c50398ac2ae828cfa471268ab507b69f2d5bbaec9ddc18b7e5e7425b3a9104b)). If those are not needed, we could have a test for this config class.